RELEASE.md

```diff
@@ -21,6 +21,13 @@
 # Release 0.16.4
 
 ## Major features and improvements
+* Added the following new datasets.
+
+| Type             | Description                                                                                             | Location                      |
+| ---------------- | ------------------------------------------------------------------------------------------------------- | ----------------------------- |
+| `MessageDataSet` | Manage email messages using [the Python standard library](https://docs.python.org/3/library/email.html) | `kedro.extras.datasets.email` |
+
+* Fixed a bug for using `ParallelRunner` on Windows.
```

Comment:
There seems to be more changes in this file than just the addition of `MessageDataSet`. Is it intentional?

Reply:
Yes, this is intentional. When trying to figure out the format for mentioning new datasets in the release notes, I saw it was inconsistent. I took the liberty to clean up a number of inconsistencies with this PR (also, in dataset implementations, since I was comparing across them for writing this in an effort to be most aligned).

If it's preferable to refactor these into a separate PR, that's possible (+1 contribution :joy:).